src/classes/UTIL_Http.cls

```diff
@@ -0,0 +1,212 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Utilities
+* @description Contains common HttpRequest and HttpResponse constants and methods
+*/
+public without sharing class UTIL_Http {
```

Comment:
It seems odd that the class is `without sharing`, but all the inner wrapper classes are `with sharing`. As a utility class, everything here should be `without sharing` or `inherited sharing` - especially when the class has no sobject interactions.

In general, inner classes should inherit the sharing setting of the parent class unless they explicitly need to be different, such as querying/manipulating a system or protected object. Where a class has no sobject interactions, it's better to leave the class as inherited sharing in case it calls out to another class that does query or modify objects - and relies on the parent class to set the context. One thing that is important is the trigger context should almost always operate in a system (without sharing) context. That is the default behavior of all apex triggers. 

The overall concept is that most trigger/domain/utility classes should be "Inherited Sharing", while UI controllers should always be "With Sharing". The advantage of this model is that the correct context is applied by the calling code. UI's will use With Sharing and that will carry down the line (until it hits a class that explicitly overrides that to "Without Sharing") while triggers will always operate as "Without Sharing" which is the expected default behavior. There are valid business cases where a trigger needs to change the context to "With Sharing", but that's quite rare.

@RadSFDeveloper

Reply:
The `UTIL_Http` should be `without sharing`  or `inherited sharing`. The inner classes should not have sharing specified.

Reply:
The inner classes are updated and sharing is removed